homeassistant/components/knx/datetime.py

```diff
@@ -41,40 +49,56 @@ async def async_setup_entry(
 ) -> None:
     """Set up entities for KNX platform."""
     knx_module = hass.data[KNX_MODULE_KEY]
-    config: list[ConfigType] = knx_module.config_yaml[Platform.DATETIME]
-
-    async_add_entities(
-        KNXDateTimeEntity(knx_module, entity_config) for entity_config in config
+    platform = async_get_current_platform()
+    knx_module.config_store.add_platform(
+        platform=Platform.DATETIME,
+        controller=KnxUiEntityPlatformController(
+            knx_module=knx_module,
+            entity_platform=platform,
+            entity_class=KnxUiDateTimeEntity,
+        ),
     )
 
-
-def _create_xknx_device(xknx: XKNX, config: ConfigType) -> XknxDateTimeDevice:
+    entities: list[KnxYamlEntity | KnxUiEntity] = []
+    if yaml_platform_config := knx_module.config_yaml.get(Platform.DATETIME):
+        entities.extend(
+            KnxYamlDateTimeEntity(knx_module, entity_config)
+            for entity_config in yaml_platform_config
+        )
+    if ui_config := knx_module.config_store.data["entities"].get(Platform.DATETIME):
+        entities.extend(
+            KnxUiDateTimeEntity(knx_module, unique_id, config)
+            for unique_id, config in ui_config.items()
+        )
+    if entities:
+        async_add_entities(entities)
+
+
+def _create_xknx_device(
+    xknx: XKNX,
+    name: str,
+    group_address: GroupAddressesType,
+    group_address_state: GroupAddressesType,
+    respond_to_read: bool,
+    sync_state: bool | float | str,
+) -> XknxDateTimeDevice:
     """Return a XKNX DateTime object to be used within XKNX."""
     return XknxDateTimeDevice(
         xknx,
-        name=config[CONF_NAME],
+        name=name,
         localtime=False,
-        group_address=config[KNX_ADDRESS],
-        group_address_state=config.get(CONF_STATE_ADDRESS),
-        respond_to_read=config[CONF_RESPOND_TO_READ],
-        sync_state=config[CONF_SYNC_STATE],
+        group_address=group_address,
+        group_address_state=group_address_state,
+        respond_to_read=respond_to_read,
+        sync_state=sync_state,
     )
 
 
-class KNXDateTimeEntity(KnxYamlEntity, DateTimeEntity, RestoreEntity):
+class _KNXDateTimeEntity(DateTimeEntity, RestoreEntity):
```

Comment:
Nitpick: Other platforms don't add "Entity" for their platform classes. For consistency, consider to use `_KnxDateTime` as class name. (same for `KnxYamlDateTime` and `KnxUiDateTime`).